rfcs/convergence/props-consistency-check.md

```diff
@@ -0,0 +1,391 @@
+# Consistency Check
+
+The following components and their props were reviewed as of 9/7/2021 for consistency.
+
+Accordian, AccordianHeader, AccordianItem, AccordianPanel
+Avatar
+Badge, CounterBadge, PresenceBadge,
+Button, CompoundButton, MenuButton, ToggleButton,
+Divider,
+Image,
+Label,
+Link,Menu, MenuDivider, MenuGroup, MenuGroupHeader,MenuItem, MenuItemCheckbox, MenuItemRadio, MenuList, MenuPopover, MenuTrigger
+Popover, PopoverSurface, PopoverTrigger
+Portal
+
+## Nit: Explicit component type declaration
+
+> Some components declare their type explicity, others do not.
+
+```tsx
+export const Accordion: React.FunctionComponent<AccordionProps & React.RefAttributes<HTMLElement>> = React.forwardRef<
+  HTMLElement,
+  AccordionProps
+>((props, ref) =>
+```
+
+```tsx
+export const Avatar = React.forwardRef((props: AvatarProps, ref: React.Ref<HTMLElement>) =>
+```
+
+## Props declared as interfaces vs types
+
+> Some props are defined as interfaces and extended. Others are defined as types and intersected. This isn't always consistent within the same component.
+
+Note: I won't list out every component here. We can scrub once we decide the right approach.
+
+```ts
+export interface AccordionProps extends ComponentProps<AccordionSlots>, Partial<AccordionCommons>
+
+export type AccordionSlots
+export interface AccordionCommons
+```
+
+```ts
+export interface AccordionHeaderProps extends ComponentProps<AccordionHeaderSlots>, Partial<AccordionHeaderCommons>
+
+export type AccordionHeaderSlots
+export interface AccordionHeaderCommons
+```
+
+## Props extending ComponentProps vs. ComponentPropsCompat
+
+> Some props are still extended from ComponentPropsCompat
+
+AvatarProps, BadgeProps, ButtonProps, ...
+
+> Some props have properties using ShorthandPropsCompat
+
+AvatarProps (badge, image, label, icon), BadgeProps (icon), ButtonProps (icon), ...
+
+## Slots and HTML attributes
+
+> Only Accordian and Menu components use ComponentProps<> with slots currently.
```

Comment:
from @ecraig12345 in HackMD:
Maybe this should be obvious but I'm not totally clear on what you mean here, sorry. Can you give an example of what it looks like in a component that doesn't use this?

Reply:
This isn't a consistency problem per se.  I think we want to make sure that the types of each of the slots make sense.  As the components using the ComponentPropsCompat are converted to using ComponentProps and start having slots then we will want to check the consistency.